website/content/docs/nia/usage/zscaler-application-segments.mdx

```diff
@@ -0,0 +1,404 @@
+  description = "This task dynamically updates application segments domain names, tcp ports"
+  providers = ["zpa"]
+  module = "zscaler/application-segment-nia/zpa"
+  condition "services" {
+    names = ["web"]
+  }
+  variable_files = [""]
+}
+```
+
+</CodeBlockConfig>
+
+The `driver` block defines all Terraform providers required to execute the task. In the following example, the `zscaler/zpa` provider is required.
+
+<CodeBlockConfig filename="tasks.hcl" hideClipboard highlight="5">
+
+```hcl
+driver "terraform" {
+  log = true
+  required_providers {
+    zpa = {
+      source = "zscaler/zpa"
+      version = "2.7.9"
+    }
+  }
+}
+```
+
+</CodeBlockConfig>
+
+The `terraform_provider` specifies the parameters to interact with the ZPA.
+
+<Note title="Security Note">
+
+   In a production environment, we recommend separating any sensitive credentials from the CTS configuration files and loading them dynamically using environment variables, Consul KV, or HashiCorp Vault. Refer to the [Terraform Registry](https://registry.terraform.io/providers/zscaler/zpa/latest/docs#authentication) for a full set of provider arguments.
+
+</Note>
+
+<CodeBlockConfig filename="tasks.hcl" hideClipboard>
+
+```hcl
+terraform_provider "zpa" {
+  zpa_client_id       = "{{ env \"ZPA_CLIENT_ID\" }}"
+  zpa_client_secret   = "{{ env \"ZPA_CLIENT_SECRET\" }}"
+  zpa_customer_id     = "{{ env \"ZPA_CUSTOMER_ID\" }}"
+}
+```
+
+</CodeBlockConfig>
+
+The `task` block defines a task for CTS to run when specific Consul services change.
+
+- `name` defines the task name.
+- `providers` defines the providers this task uses. In this case, the previously defined `zpa` provider.
+- `module` defines the path to the ZPA Terraform NIA module. This module lets Terraform dynamically automatically create and update application segments when there are changes to specific Consul services.
+- `condition "services"` defines the Consul services, when changed, would trigger this CTS task. The names must match the name of the Consul service you want to trigger the task (in this case, `web`).
+
+<CodeBlockConfig filename="tasks.hcl" hideClipboard>
+
+```hcl
+task {
+  name = "zpa_application_segment"
+  description = "This task dynamically updates application segments domain names, tcp ports"
+  providers = ["zpa"]
+  module = "zscaler/application-segment-nia/zpa"
+  condition "services" {
+    names = ["web"]
+  }
+  variable_files = [""]
+}
+```
+
+</CodeBlockConfig>
+
+For more details about the configuration, refer to the [CTS configuration docs](/consul/docs/nia/configuration). For more details on the Terraform NIA module for ZPA application segments, refer to the [Terraform Registry documentation ](https://registry.terraform.io/modules/zscaler/application-segment-nia/zpa/latest) or [GitHub repository](https://github.com/zscaler/terraform-zpa-application-segment-nia).
+
+## Start CTS
+
+Run the `consul-terraform-sync start` command to start CTS and perform the following operations:
```

Comment:
From the statement it seems that the user needs to do the three steps.

Shall we make it  explicit that these steps are automatically performed by CTS at startup?